lib/core/kotekanTrackers.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef KOTEKAN_TRACKERS_HPP
+#define KOTEKAN_TRACKERS_HPP
+
+#include "restServer.hpp" // for connectionInstance, restServer
+#include "visUtil.hpp"    // for StatTracker
+
+#include <map>      // for map
+#include <memory>   // for shared_ptr
+#include <mutex>    // for mutex
+#include <stddef.h> // for size_t
+#include <string>   // for string
+
+namespace kotekan {
+namespace trackers {
+
+class KotekanTrackers {
+
+public:
+    /**
+     * @brief Returns an instance of the kotekan trackers.
+     *
+     * @return Returns the kotekan trackers instance.
+     */
+    static KotekanTrackers& instance();
+
+    /**
+     * @brief Registers this class with the REST server, creating the
+     *        /trackers end point
+     * @param rest_server The server to register with.
+     */
+    void register_with_server(restServer* rest_server);
+
+    /**
+     * @brief The call back function for the REST server to use.
+     *
+     * This function is never called directly.
+     *
+     * @param conn The connection instance to send results to.
+     */
+    void trackers_callback(connectionInstance& conn);
+
+    /**
+     * @brief Adds a new tracker
+     *
+     * @param name The name of the tracker.
+     * @param unit The unit of the tracker.
+     * @param size The size of the tracker with default of 100.
+     * @param is_optimized The switch of min/max methods of the tracker.
+     * @return a shared pointer to the newly created tracker
+     * @throw std::runtime_error if the tracker with that name is already registered.
+     */
+    std::shared_ptr<StatTracker> add_tracker(std::string name, std::string unit, size_t size = 100,
```

Comment:
We should also have a call to remove the tracker.  And see my comment in the function regarding changing the required arguments here.

Reply:
The remove() function should remove individual tracker or all trackers in a given stage?

Reply:
I'd give the option for both.